runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -7722,58 +7734,50 @@ J9::X86::TreeEvaluator::VMnewEvaluator(
       {
       if (node->getOpCodeValue() == TR::newarray)
          {
-         if (comp->getOption(TR_DisableAllocationInlining))
-            return 0;
-
          elementSize = TR::Compiler->om.getSizeOfArrayElement(node);
          }
       else
          {
          // Must be TR::anewarray
          //
-         if (comp->getOption(TR_DisableAllocationInlining))
-            return 0;
-
-         if (comp->useCompressedPointers())
-            elementSize = TR::Compiler->om.sizeofReferenceField();
-         else
-            elementSize = (int32_t)TR::Compiler->om.sizeofReferenceAddress();
+         elementSize = comp->useCompressedPointers() ?
+            TR::Compiler->om.sizeofReferenceField() :
+            (int32_t)TR::Compiler->om.sizeofReferenceAddress();
 
          classReg = node->getSecondChild()->getRegister();
+
          // For dynamic array allocation, need to evaluate second child
          if (!classReg && dynamicArrayAllocation)
             classReg = cg->evaluate(node->getSecondChild());
          }
 
       isArrayNew = true;
 
-      if (generateArraylets)
-         {
-         dataOffset = fej9->getArrayletFirstElementOffset(elementSize, comp);
-         }
-      else
-         {
-         dataOffset = TR::Compiler->om.contiguousArrayHeaderSizeInBytes();
-         }
+      dataOffset = generateArraylets ?
+         fej9->getArrayletFirstElementOffset(elementSize, comp) :
+         TR::Compiler->om.contiguousArrayHeaderSizeInBytes();
       }
 
    TR::LabelSymbol *startLabel = generateLabelSymbol(cg);
    TR::LabelSymbol *fallThru = generateLabelSymbol(cg);
    startLabel->setStartInternalControlFlow();
    fallThru->setEndInternalControlFlow();
 
+   static char *enableTLHBatchClearing = feGetEnv("TR_EnableBatchClear");
```

Comment:
I would suggest using the `TR_J9VMBase::tlhHasBeenCleared()` query instead; it returns true if the env var is defined but also if `-Xgc:batchClearTLH` is used.

Reply:
Unless I'm missing something, that FE query always returns false currently -> https://github.com/eclipse-openj9/openj9/blob/0e34d4c87aebd0abd57dd49a2d7af6f27b726c82/runtime/compiler/env/VMJ9.cpp#L4140

But I agree that it would be better to check that.  I will change the query to check the environment variable.

Reply:
I misread the code.  But I will fix up the query to handle the environment variable as well.

Reply:
Yeah, ideally we should just drop the env var altogether. It may have been added before the GC option existed.